src/crypto/backend/rust.rs

```diff
@@ -0,0 +1,110 @@
+//! This backend implements the primitives using the [RustCrypto] ecosystem.
+//!
+//! [RustCrypto]: https://github.com/RustCrypto
+
+use alloc::vec::Vec;
+
+use digest::Digest as _;
+use rand_core::RngCore as _;
+use thiserror::Error;
+
+use super::interface;
+
+pub(crate) mod ec;
+pub(crate) mod hmac;
+pub(crate) mod okp;
+pub(crate) mod rsa;
+
+// TODO: remove the `cfg_attr` once the RustCrypto crates implement
+// the core::error::Error trait.
+
+/// The errors that can be produced by the rust crypto backend.
+#[derive(Debug, Error)]
+pub(crate) enum BackendError {
+    /// The error returned if the key is invalid.
+    #[error("invalid key length")]
+    InvalidLength,
+
+    /// RSA operation failed.
+    #[cfg_attr(feature = "std", error("an RSA operation failed"))]
+    #[cfg_attr(not(feature = "std"), error("an RSA operation failed: {0}"))]
+    Rsa(#[cfg_attr(feature = "std", source)] ::rsa::errors::Error),
+
+    /// Error of the `elliptic_curve` crate.
+    #[cfg_attr(feature = "std", error("an elliptic curve operation failed"))]
+    #[cfg_attr(not(feature = "std"), error("an elliptic curve operation failed: {0}"))]
+    EllipticCurve(#[cfg_attr(feature = "std", source)] ::elliptic_curve::Error),
+
+    /// Error of the `ecdsa` crate.
+    #[cfg_attr(feature = "std", error("an ECDSA operation failed"))]
```

Comment:
This will soon be fixed, because `signature::Error` (used in ecdsa) switched to `core::error::Error` on main branch https://github.com/RustCrypto/traits/blob/5d3e03178b822f13e85ce46316e1f524ed7921c4/signature/src/error.rs#L86